File: backend/src/models/Macro.js
import mongoose from 'mongoose'

import createSchema from './utils/createSchema'
import {Node} from './Workflow'

const MacroSchema = createSchema({
  userId: {type: String, required: true, index: true},
  name: {type: String, required: true, unique: true},
  keywords: [String],
  queryType: {type: String, required: true},
  cell: {type: Node, required: true},
  workflowNodes: {type: Map, of: Node, required: true},
})

const Macro = mongoose.model('Macro', MacroSchema)

export default Macro

File: backend-v2/internal/models/macro.go
package models

import (
	"time"
)

type Macro struct {
	ID            string          `json:"_id,omitempty" bson:"_id,omitempty"`
	UserID        string          `json:"userId" bson:"userId"`
	Name          string          `json:"name" bson:"name"`
	Keywords      []string        `json:"keywords,omitempty" bson:"keywords,omitempty"`
	QueryType     string          `json:"queryType" bson:"queryType"`
	Cell          Node            `json:"cell" bson:"cell"`
	WorkflowNodes map[string]Node `json:"workflowNodes" bson:"workflowNodes"`
	UpdatedAt     time.Time       `json:"updatedAt" bson:"updatedAt"`
	CreatedAt     time.Time       `json:"createdAt,omitempty" bson:"createdAt,omitempty"`
}
